plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -171,17 +181,36 @@ public IcebergMetadata(
     @Override
     public IcebergTableHandle getTableHandle(ConnectorSession session, SchemaTableName tableName)
     {
+        return getTableHandle(session, tableName, Optional.empty(), Optional.empty());
+    }
+
+    @Override
+    public IcebergTableHandle getTableHandle(
+            ConnectorSession session,
+            SchemaTableName tableName,
+            Optional<ConnectorTableVersion> startVersion,
+            Optional<ConnectorTableVersion> endVersion)
+    {
+        if (startVersion.isPresent()) {
+            throw new TrinoException(NOT_SUPPORTED, "Read table with start version is not supported");
+        }
+
         IcebergTableName name = IcebergTableName.from(tableName.getTableName());
         verify(name.getTableType() == DATA, "Wrong table type: " + name.getTableNameWithType());
 
+        if (name.getSnapshotId().isPresent() && endVersion.isPresent()) {
```

Comment:
This `SQL`  syntax related exception is thrown at a rather low-level.
Can this inconsistency be catched in the analyzer before passing it further to the iceberg metadata ?

Reply:
the iceberg time travel using dollar sign in table name is not generic to all connectors, I don't think that should be checked in the analyzer.

Reply:
Correct me if I'm wrong, but shouldn't the metadata class be independent of SQL ?

Reply:
`tableName$snapshotId` is treated as a table name in the analyzer, and then considered as `tableName FOR VERSION AS OF snapshotId` here, so people can say `my_table$1 FOR VERSION AS OF 2` which is accepted by SQL but should be rejected at connector metadata level.